File: app/src/main/java/com/example/footapp/model/Item.java
package com.example.footapp.model;

import java.io.Serializable;

public class Item implements Serializable {
    private int id;
    private String name;
    private int price;
    private int amount;
    private String imgUrl;
    private int type;

    public Item() {
    }

    public Item(int id, String name, int price, int amount, String imgUrl, int type) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.amount = amount;
        this.imgUrl = imgUrl;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }

    public String getImgUrl() {
        return imgUrl;
    }

    public void setImgUrl(String imgUrl) {
        this.imgUrl = imgUrl;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }
}

File: app/src/main/java/com/example/footapp/model/Item.kt
package com.example.footapp.model

import android.os.Parcelable
import com.example.footapp.Response.CategoryResponse
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize


@Parcelize
data class Item(
    @SerializedName("id")
    var id: Int? = 0,
    @SerializedName("name")
    var name: String? = "",
    @SerializedName("unitPrice")
    var price: Int? = 0,
    @SerializedName("quantity")
    var amount: Int? = 0,
    @SerializedName("imageLinks")
    var imgUrl: List<String>? = listOf(),
    val category: CategoryResponse? = null
) : java.io.Serializable, Parcelable
